src/stan/mcmc/chainset.hpp

```diff
@@ -0,0 +1,567 @@
+   *
+   * @param name parameter name
+   * @return sample sd
+   */
+  double sd(const std::string& name) const { return sd(index(name)); }
+
+  /**
+   * Compute median value of specified parameter across all chains.
+   *
+   * @param index parameter index
+   * @return median
+   */
+  double median(const int index) const {
+    Eigen::MatrixXd draws = samples(index);
+    std::vector<double> sorted(draws.data(), draws.data() + draws.size());
+    std::sort(sorted.begin(), sorted.end());
+    size_t idx = static_cast<size_t>(0.5 * (sorted.size() - 1));
+    return sorted[idx];
+  }
+
+  /**
+   * Compute median value of specified parameter across all chains.
+   *
+   * @param name parameter name
+   * @return median
+   */
+  double median(const std::string& name) const { return median(index(name)); }
+
+  /**
+   * Compute maximum absolute deviation (mad) for specified parameter.
+   *
+   * Follows R implementation:  constant * median(abs(x - center))
+   * where the value of center is median(x) and the constant is 1.4826,
+   * a scale factor for asymptotically normal consistency: `1/qnorm(3/4)`.
+   * (R stats version 3.6.2)
+   *
+   * @param index parameter index
+   * @return sample mad
+   */
+  double max_abs_deviation(const int index) const {
+    Eigen::MatrixXd draws = samples(index);
+    auto center = median(index);
+    Eigen::MatrixXd abs_dev = (draws.array() - center).abs();
+    std::vector<double> sorted(abs_dev.data(), abs_dev.data() + abs_dev.size());
+    std::sort(sorted.begin(), sorted.end());
+    size_t idx = static_cast<size_t>(0.5 * (sorted.size() - 1));
+    return 1.4826 * sorted[idx];
+  }
+
+  /**
+   * Compute maximum absolute deviation (mad) for specified parameter.
+   *
+   * Follows R implementation:  constant * median(abs(x - center))
+   * where the value of center is median(x) and the constant is 1.4826,
+   * a scale factor for asymptotically normal consistency: `1/qnorm(3/4)`.
+   * (R stats version 3.6.2)
+   *
+   * @param name parameter name
+   * @return sample mad
+   */
+  double max_abs_deviation(const std::string& name) const {
+    return max_abs_deviation(index(name));
+  }
+
+  /**
+   * Compute the quantile value of the specified parameter
+   * at the specified probability.
+   *
+   * Throws exception if specified probability is not between 0 and 1.
+   *
+   * @param index parameter index
+   * @param prob probability
+   * @return parameter value at quantile
+   */
+  double quantile(const int index, const double prob) const {
+    // Ensure the probability is within [0, 1]
+    if (prob < 0.0 || prob > 1.0) {
+      throw std::out_of_range("Probability must be between 0 and 1.");
+    }
+    Eigen::MatrixXd draws = samples(index);
+    std::vector<double> sorted(draws.data(), draws.data() + draws.size());
+    std::sort(sorted.begin(), sorted.end());
+    size_t idx = static_cast<size_t>(prob * (sorted.size() - 1));
```

Comment:
Combined with above: worth writing a helper that is outside this class and can be used for this and the couple median calculations